JavaFXUserInterface/src/main/java/Controllers/AssetTypeInfoController.java

```diff
@@ -161,6 +206,144 @@ public void attachEvents() {
+                    testSize = (int)testSlider.getValue();
+                }
+            });
+
+            try{
+                modelEvaluateBtn.setOnMouseClicked(mouseEvent -> {
+                    try {
+                        int from = (int) trainSlider.getValue()+1;
+                        int to = (int) trainSlider.getValue()+1+(int) testSlider.getValue();
+                        trainDataset  = DataPrePreprocessorController.getInstance().addRULCol(modelController.createInstancesFromAssets(assetDAO.getAssetsFromAssetTypeID(Integer.parseInt(assetType.getId())).subList(0, (int) trainSlider.getValue()-1)));
+                        testDataset = DataPrePreprocessorController.getInstance().addRULCol(modelController.createInstancesFromAssets(assetDAO.getAssetsFromAssetTypeID(Integer.parseInt(assetType.getId())).subList(from, to-1)));
+                    } catch (Exception exception) {
+                        exception.printStackTrace();
+                    }
+                    try {
+                        evaluateModels(mouseEvent);
+                    } catch (Exception exception) {
+                        exception.printStackTrace();
+                    }
+                });
+            }
+            catch (Exception e){}
+        }
+    }
+    /**
+     * Evaluates all models of a specific Asset Type
+     *
+     * @param mouseEvent is an event trigger to evlauate models
+     * @author Talal
+     */
+    private void evaluateModels(MouseEvent mouseEvent)  {
+        try {
+            ArrayList<String> models = modelDAO.getListOfModels();
+            trainDataset.setClassIndex(trainDataset.numAttributes() - 1);
+            testDataset.setClassIndex(testDataset.numAttributes() - 1);
+            Instances trainSet = populateDataset(trainSize);
+            Instances testSet = populateDataset(testSize);
+
+            for(int i=0;i<models.size();i++){
+                switch (models.get(i)) {
+                    case "Linear":
+                        LinearRegressionModelImpl linearRegressionModelImpl = new LinearRegressionModelImpl();
+                        Classifier model =  linearRegressionModelImpl.trainModel(trainDataset);
+                        calculateEvaluation(model,trainDataset,testDataset,1);
+
+                    case "LSTM":
+                        ModelStrategy lstmm = new LSTMModelImpl();
+                        Classifier lstm = lstmm.trainModel(trainDataset);
+                        calculateEvaluation(lstm,trainSet,testDataset,2);
+
+                        //                    -------------------------------------------------------------------------------------------
+                        //                    --------------- this part is commented until the rest of the models are added -------------
+                        //                    -------------------------------------------------------------------------------------------
```

Comment:
This part should be un-commented since the new models have been integrated

Reply:
I uncommented the part about the new models.

Reply:
As Jeremie said, these models have been integrated and are in the DB as of script V12, We should also try to be using the returned model object from the model classes we already have rather than creating new ones. It is fine for now but when we include parameters in sprint 8, creating them this way will only have the default parameters and not the ones we will be using in the actual model classes.

Reply:
I think we can tackle that during sprint 8 @Khal-Shah since as of now, we only keep 1 trained model per asset type in the database.